microsoft/testsuites/power/power.py

```diff
@@ -0,0 +1,680 @@
+                if not res or "hibernation exit" not in res.group(1):
+                    log.error(
+                        "VM resumed successfully "
+                        "but could not determine hibernation completion")
+                else:
+                    log.info("VM resumed successfully")
+                    vm_resumed = True
+                    break
+            else:
+                log.debug("VM is still resuming!")
+
+        if vm_resumed:
+            log.info("VM resume completed.")
+        else:
+            raise LisaException("VM resume did not finish.")
+
+        # Note that if you use NTP, the hardware clock is
+        # automatically synchronized to the system clock every 11 minutes,
+        # and this command is useful only at boot time
+        # to get a reasonable initial system time.
+        log.info("Waiting for RTC re-sync in 12 minutes...")
+        time.sleep(720)
+
+        log.info("Capturing the RTC timestmap")
+        after_time_stamp = _get_year(node)
+        log.debug(f"Got after_time_stamp: {after_time_stamp}")
+
+        # Verify the kernel panic, call trace or fatal error
+        res = re.findall('(call trace|fatal error)',
+                         dmesg.get_output(force_run=True), re.IGNORECASE)
+        if res:
+            log.error("Found Call Trace or Fatal error in dmesg")
+            # This is linux-next,
+            # so there is high chance to get call trace from other issue.
+            # For now, only print the error.
+        else:
+            log.info("Not found Call Trace and Fatal error in dmesg")
+        log.info("Waiting 60-second for logging sync")
+        time.sleep(60)
+
+        # Check the system log if it shows Power Management log
+        log.debug("Searching the keyword: 'hibernation entry' and 'hibernation exit'")
+        if _found_sys_log(node, 'hibernation entry') and \
+                _found_sys_log(node, 'hibernation exit'):
+            log.debug("Successfully found Power Management log in dmesg")
+        else:
+            raise LisaException("Could not find Power Management log in dmesg")
+
+        if before_time_stamp != after_time_stamp:
+            log.info(
+                "Successfully verified "
+                "the before_time_stamp was different from after_time_stamp")
+        else:
+            raise LisaException(
+                f"Did not find time synced. {before_time_stamp} to {after_time_stamp}")
+
+        controller_time_stamp = date.today().year
+        if after_time_stamp == str(controller_time_stamp):
+            log.info(
+                "Successfully verified the system date changed back to correct date")
+        else:
+            raise LisaException(
+                "Expected VM time changed back to the correct one after sync-up,"
+                f" but found {after_time_stamp}")
+
+        # Recover modified files
+        _recover_files(node, log, modified_files)
```

Comment:
this needs in the "final" section of a try...final.... Put above logic in a method, so it can be a short logic as above.

```python
try:
    _test_hibernation(...)
final:
    _recover_files(...)
```